corehq/apps/hqwebapp/templatetags/hq_shared_tags.py

```diff
@@ -682,6 +682,56 @@ def requirejs_main(parser, token):
     ]))
 
 
+@register.tag
+def webpack_main(parser, token):
+    """
+    Indicate that a page should be using Webpack, by naming the
+    JavaScript module to be used as the page's main entry point.
+
+    The base template need not specify a value in its `{% webpack_main %}`
+    tag, allowing it to be extended by templates that may or may not
+    use requirejs. In this case the `webpack_main` template variable
+    will have a value of `None` unless an extending template has a
+    `{% webpack_main "..." %}` with a value.
+    """
+    bits = token.contents.split(None, 1)
+    if len(bits) == 1:
+        tag_name = bits[0]
+        value = None
+    else:
+        tag_name, value = bits
+
+    if getattr(parser, "__saw_webpack_main", False):
+        raise TemplateSyntaxError(
+            "multiple '%s' tags not allowed (%s)" % tuple(bits))
+    parser.__saw_webpack_main = True
+
+    if value and (len(value) < 2 or value[0] not in '"\'' or value[0] != value[-1]):
+        raise TemplateSyntaxError("bad '%s' argument: %s" % tuple(bits))
+
+    # use a block to allow extension template to set requirejs_main for base
+    return loader_tags.BlockNode("__" + tag_name, NodeList([
+        WebpackMainNode(tag_name, value and value[1:-1])
+    ]))
+
+
+class WebpackMainNode(template.Node):
```

Comment:
Rather than having this be a nearly exact copy of `RequireJSMainNode`, what do you think of making it a subclass?
```py
class WebpackMainNode(RequireJSMainNode):
    pass
```
Obviously it will need to move down below where `RequireJSMainNode` is defined. Also, the one small difference (`context.dicts[-2][self.name] = f'webpack/{self.value}.js'`) will need to be accounted for in the one place where `WebpackMainNode` is instantiated in `webpack_main`.

Reply:
yeah this is a better approach. i was just writing some quick tests and didn't know what else I needed to change with the node

Reply:
Are we thinking of renaming `RequireJSMainNode` then?  To me, Webpack fails the `is-a` test here -- `WebpackMainNode` is not a `RequireJSMainNode`

Reply:
I mean, we could create a common base class, but that seems like overkill. They have identical implementation except for the class name, which probably isn't too important. Maybe we could just use a single common node type named something like `JSMainNode`?